app/models/variant.rb

```diff
@@ -4,6 +4,9 @@ class Variant < BaseVariant
   include Variant::Prices
 
   belongs_to :variant_category, optional: true
+  has_one :link, through: :variant_category
+  has_one :user, through: :link
```

Comment:
Preferring `has_one ..., through` over `delegate` as we no longer need to check for nil values when reaching `link`or `user` from the model object.